src/components/views/dialogs/ExportDialog.tsx

```diff
@@ -0,0 +1,395 @@
+                    { min, max },
+                ),
+            };
+        }
+
+        return {
+            valid: true,
+            feedback: _t("Enter a number between %(min)s and %(max)s", {
+                min,
+                max,
+            }),
+        };
+    };
+
+    const onCancel = async () => {
+        if (isExporting) setCancelWarning(true);
+        else onFinished(false);
+    };
+
+    const confirmCanel = async () => {
+        await Exporter?.cancelExport().then(() => {
+            setExportCancelled(true);
+            setExporting(false);
+            setExporter(null);
+        });
+    };
+
+    const exportFormatOptions = Object.keys(ExportFormats).map((format) => ({
+        value: format,
+        label: textForFormat(format),
+    }));
+
+    const exportTypeOptions = Object.keys(ExportTypes).map((type) => {
+        return (
+            <option key={type} value={type}>
+                { textForType(type) }
+            </option>
+        );
+    });
+
+    let messageCount = null;
+    if (exportType === ExportTypes.LAST_N_MESSAGES) {
+        messageCount = (
+            <Field
+                element="input"
+                type="number"
+                value={numberOfMessages.toString()}
+                ref={messageCountRef}
+                onValidate={onValidateNumberOfMessages}
+                label={_t("Number of messages")}
+                onChange={(e) => {
+                    setNumberOfMessages(parseInt(e.target.value));
+                }}
+            />
+        );
+    }
+
+    const sizePostFix = <span>{ _t("MB") }</span>;
+
+    if (exportCancelled) {
+        // Display successful cancellation message
+        return (
+            <BaseDialog
+                title={_t("Export Cancelled")}
+                className="mx_ExportDialog"
+                contentId="mx_Dialog_content"
+                onFinished={onFinished}
+                fixedWidth={true}
+            >
+                <p>{ _t("The export was cancelled successfully") }</p>
+
+                <DialogButtons
+                    primaryButton={_t("Okay")}
+                    hasCancel={false}
+                    onPrimaryButtonClick={onFinished}
+                />
+            </BaseDialog>
+        );
+    } else if (exportSuccessful) {
+        // Display successful export message
+        return (
+            <BaseDialog
+                title={_t("Export Successful")}
+                className="mx_ExportDialog"
+                contentId="mx_Dialog_content"
+                onFinished={onFinished}
+                fixedWidth={true}
+            >
+                <p>{ _t("Your messages were successfully exported") }</p>
+
+                <DialogButtons
+                    primaryButton={_t("Okay")}
+                    hasCancel={false}
+                    onPrimaryButtonClick={onFinished}
+                />
+            </BaseDialog>
+        );
```

Comment:
These could be fired as InfoDialogs once this one closes, keeps the bulk out of this dialog implementation.